### PR TITLE
Introduce 'os.hrtime()' and 'os.availableMemory()'

### PR DESCRIPTION
**Pull request by jwerle**
This PR also adds a few more missing porcelain functions for `os` and `process`

### CODE REVIEW

**Comment by mribbons** [comment]
@jwerle looks good. Is the windows CI issue caused by log poll changes, or is it due to new code?

**Comment by mribbons** [comment]
@jwerle just tried this on my Win32, I get the same result as CI:
```
# os.hrtime()
Uncaught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.

caught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
WARNING: Window#0 exiting with code 1
execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
```

**Comment by jwerle** [comment]
> @jwerle just tried this on my Win32, I get the same result as CI:
> 
> ```
> # os.hrtime()
> Uncaught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> 
> caught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> WARNING: Window#0 exiting with code 1
> execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> ```

wow! TIL. Okay I can work around this!

**Review by chicoxyzzy — APPROVED** [review]

**Comment by jwerle** [comment]
> @jwerle just tried this on my Win32, I get the same result as CI:
> 
> ```
> # os.hrtime()
> Uncaught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> 
> caught InvalidAccessError: Failed to execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> WARNING: Window#0 exiting with code 1
> execute 'open' on 'XMLHttpRequest': Synchronous requests from a document must not set a response type.
> ```

okay this _should_ be fixed!

File: src/desktop/main.cc [hunk 1]
```diff
@@ -499,6 +499,11 @@ MAIN {
       } catch (...) {
       }
 
+    #if defined(__APPLE__)
+      if (app.fromSSC) {
```
Comment (jwerle) [inline src/desktop/main.cc:503]: this is only on macOS desktop and when the application is launched from the `ssc` CLI. This allows us to get the exit code of the process, which is impossible from `ssc` as we launch the application with `NSWorkspace` and the actual process that gets launched is **NOT** a child process. So `wait`, `waitpid` and `waitid` will _NOT_ work here :(

**Pull request merged**